ckanext/harvest/model/__init__.py

```diff
@@ -356,7 +357,7 @@ def define_harvester_tables():
         Column('state', types.UnicodeText, default=u'WAITING'),
         Column('metadata_modified_date', types.DateTime),
         Column('retry_times', types.Integer, default=0),
-        Column('harvest_job_id', types.UnicodeText, ForeignKey('harvest_job.id')),
+        Column('harvest_job_id', types.UnicodeText, ForeignKey('harvest_job.id', ondelete='SET NULL'), nullable=True),
```

Comment:
Changing table structure would require migration script, otherwise it would not be applied to any instance.

Reply:
I agree that a schema change creates some challenges for existing databases.   However, the `ckanapi` package gives very nice support for exporting and re-importing users, packages, and organizations, so that a fresh database instance with imported users, etc is not very time consuming.     I could write up a set of instructions for the Wiki if this is useful.